src/data/finsupp/defs.lean

```diff
@@ -985,4 +985,35 @@ lemma update_eq_sub_add_single [add_group G] (f : α →₀ G) (a : α) (b : G)
   f.update a b = f - single a (f a) + single a b :=
 by rw [update_eq_erase_add_single, erase_eq_sub_single]
 
+/-! ### Declarations about `const_on` -/
+
+section const_on
+variables [has_zero M]
+
+/-- `const_on s b` is the finitely supported function with value `b` on `s` and zero otherwise. -/
+def const_on (s : finset α) (b : M) : α →₀ M :=
+{ support := if b = 0 then ∅ else s,
+  to_fun := λ i, if i ∈ s then b else 0,
+  mem_support_to_fun := λ a, by split_ifs; simp [h, h_1], }
```

Comment:
What are your thoughts on something more general like the following?
```suggestion
/-- `finsupp.indicator s b` is the finitely supported function corresponding to `set.indicator`. -/
def indicator (s : finset α) (f : α → M) : α →₀ M :=
{ support := s.filter (λ a, f a ≠ 0),
  to_fun := if x ∈ s then f x else 0,
  mem_support_to_fun := λ a, by split_ifs; simp [h, h_1], }
```
This has the nice benefit that we can then link it to `set.indicator` with lemmas.

Reply:
It looks like there is already a `finsupp.indicator`.

Reply:
I've renamed the original `indicator` to `pindicator` for now, because its input is a partially defined function just like `list.pmap`.